### PR TITLE
Add Aqua tests and resolve method ambiguities

### PR DESCRIPTION
**Pull request by asinghvi17**
This commit significantly narrows the types available for the first argument in `convert_arguments(::T1, ::Table)`.  This removes a method ambiguity I ran into when testing Shapefile.jl on my machine.

### CODE REVIEW

**Comment by asinghvi17** [comment]
CI failures are unrelated and fixed in #118.

**Comment by rafaqz** [comment]
Can we lock this in with Aqua.il tests?

**Comment by asinghvi17** [comment]
I guess?  Not sure how to test only one function though.

**Comment by rafaqz** [comment]
U mean test the whole thing! Maybe there are no other ambiguities and its fine? Otherwise sure we can merge it.

But you can ignore some functions or modules in Aqua ambiguity testing if you need to.

**Comment by asinghvi17** [comment]
There are actually no ambiguities which is pretty nice :D 

Thanks for the tip, I implemented Aqua but ignored all the deps compat stuff for now.  Tests should be green!

**Comment by asinghvi17** [comment]
Ah these tests are failing here too, I thought it was just an issue on my laptop.  Not sure what to do about these, they don't seem catalyzed by any change here.

File: test/runtests.jl [hunk 1]
```diff
@@ -3,255 +3,277 @@ using Shapefile
...
+
 const GI = GeoInterface
 
+import GeoInterfaceRecipes # for ambiguity ignore list only
 
 using Shapefile: Point, PointM, PointZ, Polygon, PolygonM, PolygonZ, Polyline,
     PolylineM, PolylineZ, MultiPoint, MultiPointM, MultiPointZ,
     MultiPatch, LineString, LinearRing, SubPolygon, Rect, Interval
 
-shp = Shapefile.Handle(joinpath("shapelib_testcases", "test.shp"))
```
Comment (rafaqz) [inline test/runtests.jl:13]: Can we revert all of these style changes, it's hard to see what's going on

**Comment by rafaqz** [comment]
Yes good to see. Weird that test has broken, I wonder what changed.

File: test/runtests.jl [hunk 2]
```diff
@@ -133,125 +136,144 @@ test_shapes = Dict(
...
-        @test GeoInterface.extent(s) isa GeoInterface.Extent
-        # This test should be moved to GeoInterface `testgeometry`
-        @test GeoInterface.ncoord(s) ==
-            (2 + GeoInterface.is3d(s) + GeoInterface.ismeasured(s)) ==
-            length(GeoInterface.coordnames(s))
-        # TODO Plotting a single measured point fails, needs fixing in GeoInterfaceRecipes.jl
-        GeoInterface.ismeasured(s) && GeoInterface.trait(s) isa GeoInterface.PointTrait || plot(s)
-    end
-    @test_broken GeoInterface.testgeometry(MultiPatch(Rect(1, 3, 2, 4), [0], [1], points, Interval(1, 4), [1, 2, 3, 4]))
+@testset "Shapefile.jl" begin
```
Comment (rafaqz) [inline test/runtests.jl:139]: Can we just untab all of this? 

There's no need to change anything except the added tests, there can be multiple test sets. 

We dont want all the git blame for the tests pointing at this PR.

Comment (asinghvi17) [reply test/runtests.jl:139]: I untabbed the tests, the diff seems a lot more limited now.

Comment (rafaqz) [reply test/runtests.jl:139]: Thanks

**Review by rafaqz — APPROVED** [review]

**Pull request merged**